### PR TITLE
Feature/fix 96

### PR DESCRIPTION
**Pull request by boazpoolman**
### What does it do?

It will do 2 things:

1. Make sure `[locale]` can be properly resolved
2. Make sure a pattern with multiple dynamic values can be properly resolved

### Why is it needed?

To fix said issues.

### How to test it?

Install and create a pattern with at least 2 dynamic values of which one is `[locale]`.
E.g. `/[locale]/[title]`

Afterwards try to update the entity and see if the path is properly created

### Related issue(s)/PR(s)

#96

### CODE REVIEW

**Review by Alexnortung — APPROVED** [review]
Looks perfect, and works as expected :)

File: packages/core/server/admin-api/services/query-layer-decorator.ts [hunk 1]
```diff
@@ -77,8 +77,12 @@ const decorator = (service: IDecoratedService) => ({
...
     }
 
-    // Generate the new path.
-    const generatedPath = await getPluginService('urlPatternService').resolvePattern(uid, { ...opts.data, id: entityId });
+    // Manually fetch the entity that's being updated.
+    // We do this becuase not all it's data is present in opts.data.
+    const entity = await service.findOne.call(this, uid, entityId);
+
+    // Generate the path.
+    const generatedPath = await getPluginService('urlPatternService').resolvePattern(uid, { ...opts.data, ...entity });
```
Comment (boazpoolman) [inline packages/core/server/admin-api/services/query-layer-decorator.ts:85]: I think we have to switch around `...opts.data` and `...entity` here.
As `opts.data` is the new data which should overwrite the old data.

File: packages/core/server/admin-api/services/query-layer-decorator.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ const decorator = (service: IDecoratedService) => ({
     // the id, can we create the URL alias entity and can we update
     // the previously created entity.
     const newEntity = await service.create.call(this, uid, { ...opts, data: opts.data });
-    const generatedPath = await getPluginService('urlPatternService').resolvePattern(uid, { ...opts.data, id: newEntity.id });
+    const generatedPath = await getPluginService('urlPatternService').resolvePattern(uid, { ...opts.data, ...newEntity });
```
Comment (boazpoolman) [inline packages/core/server/admin-api/services/query-layer-decorator.ts:37]: Same here

File: packages/core/server/admin-api/services/query-layer-decorator.ts (hunk 1 above)
Comment (Alexnortung) [reply packages/core/server/admin-api/services/query-layer-decorator.ts:85]: Ahh that makes sense, however I am not sure what old data means here as this is being done just after creation. But I think what you are saying might be technically more correct.

Comment (boazpoolman) [reply packages/core/server/admin-api/services/query-layer-decorator.ts]: Done

File: packages/core/server/admin-api/services/query-layer-decorator.ts (hunk 2 above)
Comment (boazpoolman) [reply packages/core/server/admin-api/services/query-layer-decorator.ts]: Done

**Review by Alexnortung — APPROVED** [review]
LGTM, still works as expected :)

**Pull request merged**